x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/transport/actions/put/TransportPutWatchAction.java

```diff
@@ -89,18 +91,30 @@ protected void doExecute(PutWatchRequest request, ActionListener<PutWatchRespons
             try (XContentBuilder builder = jsonBuilder()) {
                 watch.toXContent(builder, DEFAULT_PARAMS);
 
-                UpdateRequest updateRequest = new UpdateRequest(Watch.INDEX, Watch.DOC_TYPE, request.getId());
-                updateRequest.docAsUpsert(isUpdate == false);
-                updateRequest.version(request.getVersion());
-                updateRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-                updateRequest.doc(builder);
+                if (isUpdate) {
+                    UpdateRequest updateRequest = new UpdateRequest(Watch.INDEX, Watch.DOC_TYPE, request.getId());
+                    updateRequest.docAsUpsert(isUpdate == false);
```

Comment:
why call out `isUpdate == false` if you know that will return `false`?

Reply:
I guess I get that its our way of saying `!isUpdate` since the `!` operator is frowned upon for readability, but I would just be fine seeing `docAsUpsert(false)`

Reply:
removed that line, which is no more needed.